homeassistant/components/recorder/history/modern.py

```diff
@@ -560,26 +570,55 @@ def _get_start_time_state_for_entities_stmt(
     """Baked query to get states for specific entities."""
     # We got an include-list of entities, accelerate the query by filtering already
     # in the inner and the outer query.
+    if dialect_name == SupportedDialect.POSTGRESQL:
+        # Postgresql does not support index skip scan, so we need to do a
+        # lateral join to get the max last_updated_ts for each metadata_id
+        # as a group-by is too slow.
+        # https://github.com/home-assistant/core/issues/132865
+        max_metadata_id = StatesMeta.metadata_id.label("max_metadata_id")
+        max_last_updated = (
+            select(func.max(States.last_updated_ts))
+            .where(
+                (States.metadata_id == max_metadata_id)
+                & (States.last_updated_ts >= run_start_ts)
```

Comment:
The run_start_ts doesn't actually matter anymore because it will scan the index backwards.